scripts/bump-child-deps-utils.ts

```diff
@@ -0,0 +1,76 @@
+import c from 'chalk';
+import path from 'node:path';
+import { sh } from './utils';
+import { readJson, root, writeJson } from './versions';
+
+let newVersionNumber: string | undefined = undefined;
+
+/*
+ * Use `npm version` workspaces to get the latest version of the design-system we are attempting to update to
+ */
+const getNewDesignSystemVersion = (): string => {
+  const newDesignSystemVersionJSON = JSON.parse(sh(`npm version -w @cmsgov/design-system --json`));
+  return newDesignSystemVersionJSON['@cmsgov/design-system'];
+};
+
+/*
+ * Set the @cmsgov/design-system dependency to the new version in each child design system and example package
+ */
+const updateDSVersion = (json: JSON | any): JSON => {
+  const dsKey = '@cmsgov/design-system';
+  newVersionNumber = newVersionNumber ?? getNewDesignSystemVersion();
```

Comment:
It looks like `newVersionNumber` will only be `undefined` the first time it's called. On subsequent calls, it will already be defined. Would it be more appropriate to assign its value before entering the loop rather than within it?






[12:05](https://corbalt.slack.com/archives/D073JLCFBDZ/p1724256355490179)
On subsequent iterations of the foreEach, newVersionNumber will be defined.
12:06
So, why not define it before the loop, like so:
// This function kicks off everything.
export const updateChildDSAndExamples = (): void => {
  // Fetch the version number once before the loop.
  newVersionNumber = getNewDesignSystemVersion();
  
  const allOurPackages = getPackageNamesAndLocations();
  allOurPackages.forEach((childPackage) => {
    bumpPackageJsonDependencies(childPackage);
  });
};

Reply:
Good call out! Fixing.